Build Error

daemon/json.c: In function ‘json_tok_u64’:
daemon/json.c:54:2: error: size of unnamed array is negative
  BUILD_ASSERT(sizeof(l) >= sizeof(*num));
  ^
<builtin>: recipe for target 'daemon/json.o' failed
make: *** [daemon/json.o] Error 1

--- a/Makefile
+++ b/Makefile
@@ -247,7 +247,7 @@ ccan/ccan/cdump/tools/cdump-enumstr: ccan/ccan/cdump/tools/cdump-enumstr.o $(CDU
 libsecp256k1.a: secp256k1/libsecp256k1.la
 
 secp256k1/libsecp256k1.la:
-	cd secp256k1 && ./autogen.sh && ./configure --enable-static=yes --enable-shared=no --enable-tests=no --enable-experimental=yes --enable-module-ecdh=yes --libdir=`pwd`/..
+	cd secp256k1 && ./autogen.sh && ./configure CC="$(CC)" --enable-static=yes --enable-shared=no --enable-tests=no --enable-experimental=yes --enable-module-ecdh=yes --libdir=`pwd`/..
 	$(MAKE) -C secp256k1 install-exec
 
 lightning.pb-c.c lightning.pb-c.h: lightning.proto

--- a/daemon/json.c
+++ b/daemon/json.c
@@ -45,9 +45,9 @@ bool json_tok_u64(const char *buffer, const jsmntok_t *tok,
 		  uint64_t *num)
 {
 	char *end;
-	unsigned long l;
+	unsigned long long l;
 
-	l = strtoul(buffer + tok->start, &end, 0);
+	l = strtoull(buffer + tok->start, &end, 0);
 	if (end != buffer + tok->end)
 		return false;
 
@@ -55,7 +55,7 @@ bool json_tok_u64(const char *buffer, const jsmntok_t *tok,
 	*num = l;
 
 	/* Check for overflow */
-	if (l == ULONG_MAX && errno == ERANGE)
+	if (l == ULLONG_MAX && errno == ERANGE)
 		return false;
 
 	if (*num != l)

